src/library_syscall.js

```diff
@@ -1735,7 +1742,11 @@ for (var x in SyscallsLibrary) {
   }
 #if SYSCALL_DEBUG
   if (which && t.indexOf(', varargs') != -1) {
-    post += 'SYSCALLS.varargs = undefined;\n';
+    if (canThrow) {
+      post += 'finally { SYSCALLS.varargs = undefined; }\n';
```

Comment:
do we always emit a `try` if `SYSCALL_DEBUG`? (even if it's nothrow etc.?)

Reply:
No, only if canThrow is enabled.